application-templates/starter/src/components/channel-details/channel-details.jsx

```diff
@@ -44,31 +44,33 @@ const ChannelDetails = (props) => {
   const channelDetailsUpdater = useChannelDetailsUpdater();
   const handleSubmit = useCallback(
     async (formikValues, formikHelpers) => {
-      const data = formValuesToDoc(formikValues);
-      try {
-        await channelDetailsUpdater.execute({
-          originalDraft: channel,
-          nextDraft: data,
-        });
-        showNotification({
-          kind: 'success',
-          domain: DOMAINS.SIDE,
-          text: intl.formatMessage(messages.channelUpdated, {
-            channelName: formatLocalizedString(formikValues, {
-              key: 'name',
-              locale: dataLocale,
-              fallbackOrder: projectLanguages,
+      if (channel) {
```

Comment:
Yes, that is correct, it exists only because channel can be `undefined`.
I might change that to:
```ts
await channelDetailsUpdater.execute({
          originalDraft: channel || ({} as TChannel),
          nextDra
```
if that works better

Reply:
If I correctly follow the flow of that component, the `channelDetailsUpdate.execute` function is called within the `onSubmit` callback which can only be used when the form has been rendered and, therefore, the `channel` must exists at that point so perhaps we could also use the [non-null assertion operator](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#non-null-assertion-operator-postfix-) here for clarity.

```
await channelDetailsUpdater.execute({
          originalDraft: channel!,
          ...
```

Reply:
Yes, thanks Carlos, this also seems to be a valid option. I'm ok with that, only wondering if we should prefer assertions over explicit checking for `undefined` in the template. Of course I'm using some myself in other places but my take on that is the less the better

Reply:
Sure. There are different ways of approaching this kind of situations and I know perhaps the use of non-null operator is controversial.
It makes sense to me using in such situations as I understand its purpose to be let typescript know about a value to actually exists in context where it automatically can't detect it.

However, I would also be comfortable with other solutions we think they would be better for _educational_ purposes if we have a consensus.

Reply:
👍 updated in 51c02b97edaf5d6f10ad7f99898aa697834b38aa